docs/proposals/docker-event-collection.md

```diff
@@ -0,0 +1,86 @@
+
+**NOTE:** Only Docker Events will be parsed from the audit log; all others will be dropped.
+
+An example event messages written to `/var/log/audit/messages`:
+
+```
+type=VIRT_CONTROL msg=audit(1504273220.029:520188): pid=14500 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:container_runtime_t:s0 msg='vm=docker.io/openshift/origin-logging-kibana@sha256:a9d309679
+cd8c1b098c696b8acef9fba401c09ed7568ad266eb32208c2ce4da9 user=? auid=4294967295 reason=api op=exec vm-pid=17235 exe=container-entrypoint hostname=logging-kibana-1-xxmr3  exe="/usr/bin/dockerd-current" hostname=? 
+addr=? terminal=? res=success'
+
+```
+
+The additional fields:
+
+|Field|Type|Definition| Example |
+|----|-----|-|------|
+|op| string|The operation or action being performed|exec
+|reason | string|The constant value 'api'|api
+|vm|string|The docker image of the container|docker.io/openshift/origin-logging-kibana@sha256:a9d309679...
+|vm-pid|int|The process id of the running container|17235
+|user|string|The username associated with the login id| 
+|auid|int64|The login id of the authenticated user |4294967295
+|exe|string|The entrypoint to the container|container-entrypoint
+|hostname|string|The name of the pod for the container|logging-kibana-1-xxmr3
+
+## Openshift Ansible Modifications
+* The `openshift_logging_fluentd` role adds an inventory variable to conditionally include docker event log collection. *Note:* Docker events will, by default, not be collected and will require users to opt in.
+* The fluentd DaemonSet is updated to include an additional volumeMount for the audit log:
+```
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: logging-fluentd
+  ...
+  spec:
+    containers:
+    ...
+    - env:
+      - name: ENABLE_AUDITLOG_COLLECTION
+        value: "true"
+    - volumeMounts
+        - mountPath: /var/log/audit
+          name: auditlog
+      volumes:
+      - hostPath:
+          path: /var/log/audit
+        name: auditlog
+
+```
+
+## Common Data Model Modifications
+The [CDM](https://github.com/ViaQ/fluent-plugin-viaq_data_model) is updated to perform transformations in ruby code. Fields defined previously are mapped directly into `docker.events` tag like:
+```
+docker:
+  event:
+    op: exec
+    reason: api
+    vm: docker.io/openshift/origin-logging-kibana@sha256:a9d309679...
+    vm-pid:  17235
+    auid:  4294967295
+    exe: container-entrypoint
+    hostname: logging-kibana-1-xxmr3
+```
+## Fluentd Modifications
+* Modify fluentd configuration to conditionally include transformation plugin from the CDM
+
+## Dependencies
+ * [ProjectAtomic Docker](https://github.com/projectatomic/docker) version 1.12.6 - Docker fork with patch for enhanced auditing
+
+# Alternative Considerations
```

Comment:
Can you provide a description of why these paths were not chosen?

Reply:
added reasons